src/maestro/cli/fastapi_serve.py

```diff
@@ -186,3 +186,108 @@ def serve_agent(
     """
     server = FastAPIServer(agents_file, agent_name)
     server.run(host, port)
+
+class WorkflowChatRequest(BaseModel):
+    """Request model for chat endpoint."""
+    prompt: str
+
+class WorkflowChatResponse(BaseModel):
+    """Response model for chat endpoint."""
+    response: str
+    workflow_name: str
+    timestamp: str
+
+class WorkflowHealthResponse(BaseModel):
+    """Response model for health endpoint."""
+    status: str
+    workflow_name: str
+    timestamp: str
+ 
+class FastAPIWorkflowServer:
+    """FastAPI server for serving Maestro workflow."""
+    
+    def __init__(self, agents_file: str, workflow_file: str):
+        """Initialize the FastAPI server.
+        
+        Args:
+            agents_file: Path to the agents YAML file
+            workfloe_file: Path to the workflow YAML file
+        """
+        self.agents_file = agents_file
+        self.workflow_file = workflow_file
+        self.workflow = {}
+        self.app = FastAPI(
+            title="Maestro Workflow Server",
+            description="HTTP API for serving Maestro workflow",
+            version="1.0.0"
+        )
+        self._setup_routes()
+        self._load_workflow()
+        self.workflow_name = self.workflow.workflow["metadata"]["name"]
+    
+    def _setup_routes(self):
+        """Set up FastAPI routes."""
+        
+        @self.app.post("/chat", response_model=WorkflowChatResponse)
+        async def chat(request: WorkflowChatRequest):
+            """Chat with the workflow."""
+            try:
+                if not self.workflow:
+                    raise HTTPException(status_code=500, detail="No workflow loaded")
+                
+                response = await self.workflow.run(request.prompt)
+                return WorkflowChatResponse(
+                    response=str(response),
```

Comment:
```suggestion
                    response=json.dumps(response),
```
This need to be valid JSON for a front-end to parse it